packages/nuxt/src/pages/utils.ts

```diff
@@ -133,7 +136,10 @@ export function generateRoutesFromFiles (files: ScannedFile[], options: Generate
         continue
       }
 
-      const segmentName = tokens.map(({ value, type }) => type === SegmentTokenType.group ? '' : value).join('')
+      const segmentName = tokens
+        .filter(token => token.type !== SegmentTokenType.group)
+        .map(token => token.value)
+        .join('')
```

Comment:
I think the previous option is slightly faster as it has fewer iterations
```suggestion
      const segmentName = tokens.map(({ value, type }) => type === SegmentTokenType.group ? '' : value).join('')
```